modules/homa-lite/src/lib.rs

```diff
@@ -194,13 +224,31 @@ pub mod module {
 				Error::<T>::ExceededStakingCurrencyMintCap
 			);
 
-			let current_batch = Self::current_batch();
 			let staking_currency = T::StakingCurrencyId::get();
 
 			// ensure the user has enough funds on their account.
 			T::Currency::ensure_can_withdraw(staking_currency, &who, amount)?;
 
-			// Cross-chain transfers the staking assets.
+			// Calculate how much Liquid currency is to be minted.
+			// Gets the current exchange rate
+			let staking_total = Self::staking_total_issuance();
+			let exchange_rate = match staking_total.is_zero() {
+				true => Ok(T::DefaultExchangeRate::get()),
+				false => {
+					let liquid_total = T::Currency::total_issuance(T::LiquidCurrencyId::get());
+					Ratio::checked_from_rational(liquid_total, staking_total).ok_or(ArithmeticError::Overflow)
+				}
+			}?;
+
+			// liquid_to_mint = ( (staked_amount - MintFee) * liquid_total / staked_total ) * (1 -
+			// MaxRewardPerEra)
+			let mut liquid_to_mint = exchange_rate
+				.checked_mul_int(amount - T::MintFee::get())
```

Comment:
Use `checked_sub` with `.expect("proof of won't panic")`

Reply:
i was wondering in the lines above I've checked to ensure amount > (mintfee + minimum), so this subtraction is guaranteed to succeed - do I still need the checked_sub? or is this still needed for good coding style?